java/feathub-connectors/flink-connectors/flink-connector-prometheus/src/main/java/com/alibaba/feathub/flink/connectors/PrometheusSinkFunction.java

```diff
@@ -0,0 +1,351 @@
+ *                 Schema.newBuilder()
+ *                         .column(
+ *                                 "metric_map",
+ *                                 DataTypes.MAP(DataTypes.STRING(), DataTypes.INT()))
+ *                         .withComment("count_map_key=null")
+ *                         .column("metric", DataTypes.DOUBLE())
+ *                         .withComment("type=double")
+ *                         .build())
+ *         .option("serverUrl", "localhost:9091")
+ *         .option("jobName", "exampleJob")
+ *         .option("extraLabels", "k1=v1;k2=v2")
+ *         .build();
+ * </pre>
+ */
+public class PrometheusSinkFunction extends RichSinkFunction<RowData> {
+
+    private final ReadableConfig config;
+    private final String jobName;
+    private final Map<String, String> extraLabels;
+    private final List<DataType> fieldTypes;
+    private final List<MetricMeta> metricMetas;
+    private final Map<String, SimpleCollector<Gauge.Child>> nameToCollectors;
+    private CollectorRegistry registry;
+    private PushGateway pushGateway;
+
+    public PrometheusSinkFunction(ReadableConfig config, ResolvedSchema schema) {
+        this.config = config;
+        this.fieldTypes = schema.getColumnDataTypes();
+        this.nameToCollectors = new HashMap<>();
+        this.jobName = config.get(JOB_NAME);
+        this.extraLabels = parseLabels(config.get(EXTRA_LABELS));
+
+        final List<Column> columns = schema.getColumns();
+        this.metricMetas = new ArrayList<>(columns.size());
+
+        Map<String, Set<String>> metricNameToLabelNames = new HashMap<>();
+        for (Column column : columns) {
+            verifyType(column.getDataType().getLogicalType());
+            final MetricMeta metricMeta = MetricMeta.fromColumn(column);
+            final Set<String> labelNames = new HashSet<>(Arrays.asList(metricMeta.getLabelNames()));
+            if (!metricNameToLabelNames.containsKey(metricMeta.name)) {
+                metricNameToLabelNames.put(metricMeta.name, labelNames);
+            }
+            if (!metricNameToLabelNames.get(metricMeta.name).equals(labelNames)) {
+                throw new RuntimeException(
+                        String.format(
+                                "Invalid metric: %s. Metric with same name should have the same label names.",
+                                metricMeta.name));
+            }
+            metricMetas.add(metricMeta);
+        }
+    }
+
+    @Override
+    public void open(Configuration parameters) throws Exception {
+        super.open(parameters);
+        registry = new CollectorRegistry();
+        pushGateway = new PushGateway(config.get(SERVER_URL));
+
+        for (final MetricMeta metricMeta : metricMetas) {
+            final String metricName = metricMeta.name;
+            final SimpleCollector<Gauge.Child> collector =
+                    nameToCollectors.computeIfAbsent(
+                            metricName,
+                            key ->
```

Comment:
Here we use `name` (as prefix of `nameToCollectors`), `metricName` and `key` for the same thing, and then uses it as parameter of `..name(key)` below. This makes code inconsistent and hard to read.

Can you go over this PR and fix inconsistencies like this?